mlflow/models/signature.py

```diff
@@ -238,3 +245,76 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     if input_schema is None and output_schema is None:
         return None
     return ModelSignature(inputs=input_schema, outputs=output_schema)
+
+
+def set_signature(
+    model_uri: str,
+    signature: ModelSignature,
+):
+    """
+    Sets the model signature for specified model artifacts.
+
+    The process involves downloading the MLmodel file in the model artifacts (if it's non-local),
+    updating its model signature, and then overwriting the existing MLmodel file. Should the
+    artifact repository associated with the model artifacts disallow overwriting, this function will
+    fail. Furthermore, as model registry artifacts are read-only, model artifacts located in the
+    model registry and represented by ``models:/`` URI schemes are not compatible with this API.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                      - ``mlflow-artifacts:/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+                      artifact-locations>`_.
+
+                      Please note that model URIs with the ``models:/`` scheme are not supported.
+
+    :param signature: ModelSignature to set on the model.
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        from mlflow.models import set_signature, infer_signature
+
+        # load model from run artifacts
+        run_id = "96771d893a5e46159d9f3b49bf9013e2"
+        artifact_path = "models"
+        model_uri = "runs:/{}/{}".format(run_id, artifact_path)
+        model = mlflow.pyfunc.load_model(model_uri)
+
+        # determine model signature
+        test_df = ...
+        predictions = model.predict(test_df)
+        signature = infer_signature(test_df, predictions)
+
+        # set the signature for the logged model
+        set_signature(model_uri, signature)
+    """
+    assert isinstance(
+        signature, ModelSignature
+    ), "The signature argument must be a ModelSignature object"
+    assert not ModelsArtifactRepository.is_models_uri(
+        model_uri
+    ), "Model URIs with the `models:/` scheme are not supported."
```

Comment:
Can we turn these into `MlflowException` with error code `INVALID_PARAMETER_VALUE`?